### DISCUSSION TITLE
Why can't I create a new Maven project?

### DISCUSSION THREAD

**Discussion by esharris** [eclipse-m2e/m2e-core#948] [Q&A]
I decided to reinstall Eclipse because I was having trouble creating Maven archetypes while creating a Maven project.

I went to https://www.eclipse.org/downloads/

I clicked on "Download x86-64".

I downloaded the image.

I double-clicked "eclipse-inst-jre-mac64.dmg".

I dragged and dropped "Eclipse Installer" into Applications.

I went to Applications and double-clicked "Eclipse."

I launched into my workspace.

When I selected File->New->Maven Project, nothing happened.
When I selected File->New->Other ..., I got the dialog box.
When I selected Maven Project, I saw a blank "New Maven Project" dialog box.
(In previous installation tries, I would get an error message: "Cannot invoke "org.eclipse.m2e.core.ui.internal.archetype.ArchetypePlugin.getActiveArchetypeCatalogs()" because "archetypeManager" is null")

When I click "Next >," nothing happens.
I can create a standard Java project that uses Ant.

I am running on an Intel MacBook air with macOS Monterey 12.6.

**Comment by HannesWell** [comment]
Can you please make sure that your Eclipse installation uses Java-17, which is required for M2E.

**Reply by esharris** [reply]
In their build paths, each project's library includes JRE [17.0.3]. The compiler compliance level is also 17.

**Reply by HannesWell** [reply]
The question is more your Eclipse-Installation itself runs on Java-17? But unless you specified another JVM, it is likely the case because by default the JRE that is running Eclipse is used as default JDK (if it is a JDK).
Just to make sure, you can check if your eclipse.ini has a `-vm` entry that points to a jdk>=17. If no such entry is present you can check your system's default JRE, by executing `java --version` on a console.

But probably more interesting is to check if the m2e Plugins are properly installed into your installation. Can you check in the Menu `Help` -> `About Eclipse IDE` then click `Installation Details` and check what is shown in the `Installed Software` tab when you search for `m2e`.
At least `org.eclipse.m2e.feature.feature.group` should be there (and probably a few more depending on your installation).

You can also check the `Plug-ins` tab if for example `org.eclipse.m2e.core.ui` and other m2e plugins are displayed.

Another thing you could check is the Error-Log, if there are some entries that could be related.
In general I suspect that in your Eclipse either m2e is not properly installed or the plug-ins could not be activated. However it is a bit strange that a plain new installation does not work. Which package did you choose in the installer?

**Reply by esharris** [reply]
Thanks for your help.
When I just had Eclipse, “java –version” gave a message that indicated that java was not on the path. When I added Netbeans, I had to install Java 17. So the command results are now ...

java 17.0.4.1 2022-08-18 LTS
Java(TM) SE Runtime Environment (build 17.0.4.1+1-LTS-2)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.4.1+1-LTS-2, mixed mode, sharing)

“About Eclipse” is now an “Eclipse” menu item. The item org.eclipse.m2e.feature.feature.group and some other org.eclipse.m2e.* items are there.
Under the Plug-ins tab, the plug-in org.eclipse.m2e.core.ui and some other org.eclipse.m2e.* items are there.
When I try to create a Maven project and come up with a blank dialog box, the Error Log displays “Unhandled event loop exception”

To get this installation, I went to https://www.eclipse.org/downloads/. I installed “Eclipse IDE for Java Developers” and moved it to “Applications.” I selected “macOS x86_64” because I have an Intel Mac. This application has the problem.

Again, thanks for your help.

**Reply by laeubi** [reply]
Please check inside eclipse directly:
![grafik](https://user-images.githubusercontent.com/1331477/193592514-89746311-393a-4995-b0fb-00eaf5f7fbe6.png)

**Reply by esharris** [reply]
Here it is.

java. version=17.0.4.1
java.version.date=2022-08-12

**Reply by esharris** [reply]
I'm seeing the following warning for all my existing maven projects.

Description	Resource	Path	Location	Type
Project configurator "org.maven.ide.eclipse.configuration.wtp.webfragment.configurator" required by plugin execution "org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (execution: default-compile, phase: compile)" is not available. To enable full functionality, install the project configurator and run Maven->Update Project Configuration.	pom.xml	/Utilities	line 1	Maven Project Build Lifecycle Mapping Problem

The message doesn't explain how to install the project configurator. But I thought this information might help us diagnose the problem.

**Reply by HannesWell** [reply]
> I'm seeing the following warning for all my existing maven projects.
> 
> Description Resource Path Location Type Project configurator "org.maven.ide.eclipse.configuration.wtp.webfragment.configurator" required by plugin execution "org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (execution: default-compile, phase: compile)" is not available.

Does the Quick-Fix (Ctrl+1) give you an option to install that connector?
If it does but the connector fails to install with the new M2E 2.0 please ask the maintainers to update and once that update is released please create a corresponding issue at https://github.com/eclipse-m2e/m2e-discovery-catalog to include that now M2E-2 ready connector.

However not having that connector should not prevent you from creating new connectors. Unfortunately I cannot tell you much more at the moment. If you can attach the complete configuration of your Eclipse (the entire content of the 'Configuration' in the screenshot of @laeubi) maybe we can find something but I'm not too optimistic.

**Reply by esharris** [reply]
Now I'm unable to recreate the warning. I see 0 problems and no markers, but I still can't create a Maven project. This situation is frustrating; how does a problem with a project "get better?" And why am I experiencing the original problem?

**Reply by HannesWell** [reply]
Fully understand that this is frustrating, but with the provided information unfortunately I cannot help you further.
As said above maybe the full 'Configuration' of your IDE gives another hint.

**Comment by esharris** [comment]
Here is what I got from the configuration tab from the Eclipse IDE Installation Details". It's big.
[configuration.pdf](https://github.com/eclipse-m2e/m2e-core/files/9757144/configuration.pdf)

**Reply by HannesWell** [reply]
The next time, please provide it as plain txt file and just copy&paste the content from the Configuration. This makes it simpler to read long lines because they are not broken as they are in the pdf now.

However the only suspicious think I could find was that the `m2e.core.ui` Plugin is in state STARTING and not ACTIVE:
```
org.eclipse.m2e.core (2.0.3.20220904-1703) "M2E Maven Integration for Eclipse Core" [Active]
org.eclipse.m2e.core.ui (2.0.0.20220726-0935) "M2E Maven Integration for Eclipse UI" [Starting]
```
Since it has the header `Bundle-ActivationPolicy: lazy` it should actually go into active state quickly.
@laeubi could this be some DS activation deadlock again?

**Comment by laeubi** [comment]
One can got to the Console View, then open "OSGi Host Console" then enter "list" and copy the text from there to see if anything is missing, but starting can also mean that the bundles is not yet activated.

**Reply by esharris** [reply]
I don't know how to proceed. I can see the console tab. But I don't know how to open the "OSGI Host Console."

**Reply by laeubi** [reply]
![grafik](https://user-images.githubusercontent.com/1331477/196235573-bbc60aab-174a-490e-95b6-b7f3ffdeb490.png)

**Reply by esharris** [reply]
My menu displays "1 New Console View", "2 Java Stack Trace Console", and "3 Maven Console". It doesn't have "Host OSGi Console".

**Reply by laeubi** [reply]
Can you please try out the [Eclipse IDE for RCP and RAP Developers](https://www.eclipse.org/downloads/packages/release/2022-09/r/eclipse-ide-rcp-and-rap-developers)

**Reply by esharris** [reply]
Trying out this alternate Eclipse IDE doesn't fix the problem.
File -> New -> Maven Project doesn't do anything.
File -> New -> Other and selecting Maven->Maven Project or Maven -> Maven Module leads to a blank form.
Sometimes, I get ... An error has occurred. See error log for more details.
Cannot invoke "org.eclipse.m2e.core.ui.internal.archetype.ArchetypePlugin.getActiveArchetypeCatalogs()" because "archetypeManager" is null

**Reply by esharris** [reply]
I'm thinking about erasing the .eclipse directory in my home directory. The idea is to clean the next installation like I never had eclipse installed. Will that cause more harm than good? Do I need to delete any other things to eradicate the old install?

**Reply by mickaelistria** [reply]
It can be a good idea. FWIW, I personally don't use the installer but use a zipped package which I unpack in a dedicated folder; and I've been immune to this kind of issues so far and didn't miss anything. Shared .eclipse directories can be useful if you have multiple installations of Eclipse IDE; but I don't think it's a common case, nor that it is a best practice nowadays.

**Reply by esharris** [reply]
Erasing the .eclipse directory didn't fix the problem.
